src/dll/edbgEcmdDll.C

```diff
@@ -1696,11 +1702,268 @@ uint32_t dllQueryHostMemInfoRanges( const std::vector<ecmdChipTarget> & i_target
 #endif // ECMD_REMOVE_MEMORY_FUNCTIONS
 
 #ifndef ECMD_REMOVE_INIT_FUNCTIONS
+
+#ifdef EDBG_ISTEP_CTRL_FUNCTIONS
+
+static edbgIPLTable g_edbgIPLTable;
+/**
+  * @brief Set the IPL mode
+  *
+  * @param enum ipl_mode i_mode - IPL mode (Interactive or Continous IPL mode)
+  *
+  * @return Upon success, ECMD_SUCCESS will be returned.  A reason code will
+  *         be returned if the execution fails.
+  */
+uint32_t setIplMode(enum IplMode_t i_mode)
+{
+    uint32_t rc = ECMD_SUCCESS;
+    
+    /* If IPL mode is set to interactive then, SBE will be set in ISTEP mode
+     * and execute all SBE isteps in chip op mode */
+    if (i_mode == IPL_MODE_ISTEP)
+    {
+        ecmdChipTarget target;
+        constexpr uint32_t MBOX_SCRATCH_REG3 = 0x283a;
+        constexpr uint32_t MBOX_SCRATCH_REG8 = 0x283f;
+        ecmdDataBuffer dataBuffer;
+
+        target.chipTypeState = ECMD_TARGET_FIELD_VALID;
+        target.cageState = target.nodeState = target.slotState = target.posState = ECMD_TARGET_FIELD_VALID;
+        target.cage = target.node = target.slot = 0;
+
+        //This will be actually set by sbe_config_update istep procedure.
+        //For testing purpose, we are setting it up.
+        dataBuffer.setWordLength(1);
+        dataBuffer.setWord(0, 0x86000000);
+        rc = dllPutCfamRegister(target, MBOX_SCRATCH_REG3, dataBuffer);
+        dataBuffer.setWord(0, 0x20000000);
+        rc = dllPutCfamRegister(target, MBOX_SCRATCH_REG8, dataBuffer);
+        if (rc)
+        {
+            out.error(EDBG_GENERAL_ERROR, FUNCNAME,
+                      "Failed dllPutCfamRegister()\n");
+            return rc;
+        }
+    }
+    return rc;
+}
+
+/**
+  * @brief This is a helper function to call the pdbg interface
+  *        to execute an istep
+  *
+  * @param uint16_t i_start_index - position of the istep
+  * @param uint16_t i_major - Major number to execute
+  * @param uint16_t i_minor_start - Minor number start
+  * @param uint16_t i_minor_end   - Minor number end
+  *
+  * @return Upon success, ECMD_SUCCESS will be returned.  A reason code will
+  *         be returned if the execution fails.
+  */
+uint32_t dllIStepsHelper(uint16_t i_start_index, uint16_t i_major,
```

Comment:
If a function isn't actually implementing an eCMD API, it shouldn't start with dll.

Reply:
Is there a reason this is implemented in edbgEcmdDll.C and not in the edbgIsteps.C file?

Reply:
sure edbgIsteps.C is more of istep processing/validation function. since dllIStepsHelper is just a helper function that calls immediate pdbg or libipl api it makes more sense to maintained in edbgEcmdDll.C. I will remove dll from it